Upstream-Status: Inappropriate
Signed-off-by: Phil Blundell <philb at gnu.org>

Use makekeys from -native package rather than trying to compile it here, and fix up
configure logic to find keysymdef.h in the sysroot.

Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am	2012-05-24 07:13:08.989558521 +0100
+++ git/Makefile.am	2012-05-24 07:13:09.261557338 +0100
@@ -1,5 +1,5 @@
 # Order: makekeys before src
-SUBDIRS = include makekeys src test
+SUBDIRS = include src test
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xkbcommon.pc
Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2012-05-24 07:13:08.989558521 +0100
+++ git/configure.ac	2012-05-24 07:13:09.261557338 +0100
@@ -74,7 +74,8 @@
 
 # Obtain the path to the X Window System Core Protocol
 AC_MSG_CHECKING([for X11 includedir])
-includex11dir="`$PKG_CONFIG --variable=includex11dir xproto`"
+m4_pattern_allow([PKG_CONFIG_SYSROOT_DIR])
+includex11dir="$PKG_CONFIG_SYSROOT_DIR`$PKG_CONFIG --variable=includex11dir xproto`"
 AC_MSG_RESULT([$includex11dir])
 
 # Obtain full path for keysymdef header file
Index: git/src/Makefile.am
===================================================================
--- git.orig/src/Makefile.am	2012-05-24 07:13:08.993558503 +0100
+++ git/src/Makefile.am	2012-05-24 07:16:16.860681455 +0100
@@ -7,6 +7,7 @@
 libxkbcommon_la_LIBADD = xkbcomp/libxkbcomp.la
 libxkbcommon_la_LDFLAGS = -no-undefined
 libxkbcommon_la_SOURCES = \
+	$(BUILT_SOURCES) \
 	XKBcommonint.h \
 	alloc.c \
 	atom.c \
@@ -26,5 +27,5 @@
 BUILT_SOURCES = ks_tables.h
 CLEANFILES = $(BUILT_SOURCES)
 
-ks_tables.h: $(top_builddir)/makekeys/makekeys$(EXEEXT) $(KEYSYMDEF_H) $(XF86KEYSYM_H)
-	$(AM_V_GEN)cat $(KEYSYMDEF_H) $(XF86KEYSYM_H) | $(top_builddir)/makekeys/makekeys$(EXEEXT) > $@
+ks_tables.h: $(KEYSYMDEF_H) $(XF86KEYSYM_H)
+	$(AM_V_GEN)cat $(KEYSYMDEF_H) $(XF86KEYSYM_H) | libxkbcommon-makekeys$(EXEEXT) > $@
